libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -332,7 +332,10 @@ static uint8_t * encodeString( uint8_t * pDestination,
     *pBuffer = UINT16_LOW_BYTE( sourceLength );
     pBuffer++;
 
-    /* Copy the string into pBuffer. */
+    /* Copy the string into pBuffer.
+     * This memcpy intentionally copies bytes from a void * buffer into
+     * a uint8_t * buffer. */
+    /* coverity[misra_c_2012_rule_21_15_violation] */
```

Comment:
I know we had this message before, but I think we should try to have as few suppressions as possible going forward, and it looks like it might be simple to fix. Do either of the following work, or do they lead to another violation?
```
char *pDest = ( char * ) pBuffer;
( void ) memcpy( pDest, source, sourceLength);
```
or
```
const uint8_t *pSourceStr = ( const uint8_t * ) source;
( void ) memcpy( pBuffer, pSourceStr, sourceLength );
```

Reply:
Addressed it. Thanks for the suggestions